### PR TITLE
Combine the Roadmap / Quarterly Roadmap sections

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of #7013 

## Rationale for this change

The contributor guide is getting long  so let's try and combine some of the sections

## What changes are included in this PR?

Combine the Roadmap / Quarterly Roadmap sections

Screenshot 2024-07-12 at 6 34 29 AM

## Are these changes tested?
By Ci and I tested it locally as well

## Are there any user-facing changes?
Just docs

### CODE REVIEW

File: docs/source/contributor-guide/roadmap.md [hunk 1]
```diff
@@ -43,3 +43,84 @@ start a conversation using a github issue or the
 make review efficient and avoid surprises.
 
 [The current list of `EPIC`s can be found here](https://github.com/apache/datafusion/issues?q=is%3Aissue+is%3Aopen+epic).
+
+# Quarterly Roadmap
+
+A quarterly roadmap will be published to give the DataFusion community
```
Comment (alamb) [inline docs/source/contributor-guide/roadmap.md:49]: I slightly reworded this introduction section; Otherwise the content is the same

**Review by wjones127 — APPROVED** [review]

File: docs/source/contributor-guide/roadmap.md (hunk 1 above)
Comment (alamb) [reply docs/source/contributor-guide/roadmap.md:49]: That is an excellent question. I started a discussion here https://github.com/apache/datafusion/issues/11442 to see what we can find out

**Comment by alamb** [comment]
Thank you @comphead  and @wjones127

**Comment by alamb** [comment]
Since this PR just moves some content I will merge it in. Let's have the roadmap discussion on https://github.com/apache/datafusion/issues/11442

**Pull request merged**